File: gameserver/network/packets/ping.py
import time

from gameserver.game.vector import Vector
from gameserver.network.packet import Packet
from gameserver.network.packets.pong import PongPacket
from gameserver.network.utils.writer import Writer


class PingPacket(Packet):
    PACKET_ID = 1007

    def __init__(self):
        super().__init__()
        self.packet_id = self.PACKET_ID

    def parse_packet(self):
        pass

    def handle_packet(self, client):
        """On Received Ready Packet"""
        print("Ping Packet Received")
        client.last_ping_time = time.time()
        pong_packet = PongPacket()
        client.send(pong_packet)

    def finalize(self):
        writer = Writer(self.packet_id)
        return writer.finalize()

File: src/network/packets/ping.js
import Packet from '../packet.js';
import Writer from '../utils/writer.js';
import Player from "../../ui/objects/player.js";
import {convertIntColorToHex} from "../../ui/utils.js";
import Point from "../../ui/objects/point";


class PingPacket extends Packet {

    constructor() {
        super();
        this.packetId = 1007;
    }


    // Handel Server Response
    parsePacket() {

    }

    finalize() {
        // Handle Server Request
        const writer = new Writer(this.packetId);
        return writer.finalize();
    }


    handleReceivedPacket(client) {
    }
}

export default PingPacket;